docs/ARCHITECTURE.md

```diff
@@ -61,4 +61,47 @@ When a table is able to run repair the RepairState calculates the next tokens to
 The calculation is performed by the [VnodeRepairGroupFactory](../core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/state/VnodeRepairGroupFactory.java) by default.
 The TableRepairJob then generates [RepairGroups](../core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/RepairGroup.java) which are snapshots from how the state was when it was calculated.
 When the RepairGroup is executed it will generate one [RepairTask](../core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/RepairTask.java) per token range to repair.
-The RepairTask is the class that will perform the repair.
\ No newline at end of file
+The RepairTask is the class that will perform the repair.
+
+## Sub-range repairs
+
+As of #96 the repair scheduler in ecChronos has support for sub range repairs within virtual nodes.
+This is activated by specifying a target repair size in the [RepairConfiguration](../core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/RepairConfiguration.java).
+For the standalone version the option is called `repair.size.target` in the configuration file.
+Each sub-range repair session will aim to handle the target amount of data.
+
+*Note: Without this option specified the repair mechanism will handle full virtual nodes only (including how it interprets the repair history)*  
+*Note: The target repair size is assuming a uniform data distribution across partitions on the local node*
+
+### Example  
```

Comment:
Yes and no.. The problem is that the algorithm kind of assumes that each token in the ring contains roughly the same amount of data. I don't think it's possible (without looking at the data) to decide how much each token actually contains.

The full mechanism is just an abstraction layer for the user where you specify a size-based "token range size" rather than specifying how many tokens you want to repair in each session. This also gives a nice benefit in that you can specify a standard amount of data you want to handle in each repair session so that small tables automatically gets repaired in fewer sessions while larger tables get split up more.